File: Sakila_Net/Sakila_Net/VB_SakilaLib/Model/Payment.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClPayment

   private paymentId as Integer
   private  customerId  as ClCustomer
   private  staffId  as ClStaff
   private  rentalId  as ClRental
   private amount as Integer
   private paymentDate as Date
   private lastUpdate as Date


   public Property DataPaymentId() as Integer
        Get
            Return paymentId
        End Get
        Set(ByVal Value as Integer)
            paymentId = Value
        End Set        
   End Property   
   public Property DataCustomerId()  as ClCustomer
        Get
            Return customerId
        End Get
        Set(ByVal Value  as ClCustomer)
            customerId = Value
        End Set        
   End Property   
   public Property DataStaffId()  as ClStaff
        Get
            Return staffId
        End Get
        Set(ByVal Value  as ClStaff)
            staffId = Value
        End Set        
   End Property   
   public Property DataRentalId()  as ClRental
        Get
            Return rentalId
        End Get
        Set(ByVal Value  as ClRental)
            rentalId = Value
        End Set        
   End Property   
   public Property DataAmount() as Integer
        Get
            Return amount
        End Get
        Set(ByVal Value as Integer)
            amount = Value
        End Set        
   End Property   
   public Property DataPaymentDate() as Date
        Get
            Return paymentDate
        End Get
        Set(ByVal Value as Date)
            paymentDate = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (paymentId    as Integer,customerId  as ClCustomer,staffId  as ClStaff,rentalId  as ClRental,amount    as Integer,paymentDate    as Date,lastUpdate    as Date)
		   dataPaymentId = paymentId
		   dataCustomerId = customerId
		   dataStaffId = staffId
		   dataRentalId = rentalId
		   dataAmount = amount
		   dataPaymentDate = paymentDate
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (paymentId    as Integer)
    dataPaymentId = paymentId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "paymentId" & chr(34) & ":" & chr(34) & dataPaymentId & chr(34) _
		    & "," & chr(34) & "customerId" & chr(34) & ":" & chr(34) & dataCustomerId.ToString() & chr(34) _
		    & "," & chr(34) & "staffId" & chr(34) & ":" & chr(34) & dataStaffId.ToString() & chr(34) _
		    & "," & chr(34) & "rentalId" & chr(34) & ":" & chr(34) & dataRentalId.ToString() & chr(34) _
		    & "," & chr(34) & "amount" & chr(34) & ":" & chr(34) & dataAmount & chr(34) _
		    & "," & chr(34) & "paymentDate" & chr(34) & ":" & chr(34) & dataPaymentDate & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Php/Entities/Payment.php
<?PHP
class Payment implements JsonSerializable{

   private $paymentId;
   private $amount;
   private $paymentDate;
   private $lastUpdate;


   public function getPaymentId(){
        return $this->paymentId;
   }
   public function setPaymentId($paymentId){
        $this->paymentId = $paymentId;
   }
   
   public function getCustomerId(){
        return $this->customerId;
   }
   public function setCustomerId($customerId){
        $this->customerId = $customerId;
   }
   
   public function getStaffId(){
        return $this->staffId;
   }
   public function setStaffId($staffId){
        $this->staffId = $staffId;
   }
   
   public function getRentalId(){
        return $this->rentalId;
   }
   public function setRentalId($rentalId){
        $this->rentalId = $rentalId;
   }
   
   public function getAmount(){
        return $this->amount;
   }
   public function setAmount($amount){
        $this->amount = $amount;
   }
   
   public function getPaymentDate(){
        return $this->paymentDate;
   }
   public function setPaymentDate($paymentDate){
        $this->paymentDate = $paymentDate;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function Payment($paymentId,$customerId='',$staffId='',$rentalId='',$amount='',$paymentDate='',$lastUpdate=''){
		   $this->setPaymentId($paymentId);
		   $this->setCustomerId($customerId);
		   $this->setStaffId($staffId);
		   $this->setRentalId($rentalId);
		   $this->setAmount($amount);
		   $this->setPaymentDate($paymentDate);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($paymentId,$customerId='',$staffId='',$rentalId='',$amount='',$paymentDate='',$lastUpdate='') {
		   $this->setPaymentId($paymentId);
		   $this->setCustomerId($customerId);
		   $this->setStaffId($staffId);
		   $this->setRentalId($rentalId);
		   $this->setAmount($amount);
		   $this->setPaymentDate($paymentDate);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "paymentId" . chr(34) . ":" . chr(34) . $this->getPaymentId() . chr(34) 
		    . "," . chr(34) . "customerId" . chr(34) . ":" . chr(34) . $this->getCustomerId() . chr(34) 
		    . "," . chr(34) . "staffId" . chr(34) . ":" . chr(34) . $this->getStaffId() . chr(34) 
		    . "," . chr(34) . "rentalId" . chr(34) . ":" . chr(34) . $this->getRentalId() . chr(34) 
		    . "," . chr(34) . "amount" . chr(34) . ":" . chr(34) . $this->getAmount() . chr(34) 
		    . "," . chr(34) . "paymentDate" . chr(34) . ":" . chr(34) . $this->getPaymentDate() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new Payment();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>
